---
title: "Ballgown analysis"
output: html_notebook
---

```{r}
#load packages
library(ballgown)
library(RSkittleBrewer)
library(genefilter)
library(dplyr)
library(devtools)
```

```{r}
#load phenotype data
pheno_data = read.csv("geuvadis_phenodata.csv")
head(pheno_data)
```

```{r}
#load expression data
bg_chrX = ballgown(dataDir = "ballgown", samplePattern = "ERR", pData=pheno_data)
```

```{r}
#remove low adundance genes
bg_chrX_filt = subset(bg_chrX,"rowVars(texpr(bg_chrX)) >1",genomesubset=TRUE)
```

```{r}
#identify transcripts that show statistically detectable differences between groups
results_transcripts = stattest(bg_chrX_filt, feature="transcript",covariate="sex",adjustvars =
c("population"), getFC=TRUE, meas="FPKM")

head(results_transcripts)
```

```{r}
#identify genes that show statistically detectable differences between groups
results_genes = stattest(bg_chrX_filt, feature="gene",
covariate="sex", adjustvars = c("population"), getFC=TRUE,
meas="FPKM")

head(results_genes)
```

```{r}
#add gene names and gene IDs to the "results_transcripts" dataframe
results_transcripts =
data.frame(geneNames=ballgown::geneNames(bg_chrX_filt),
geneIDs=ballgown::geneIDs(bg_chrX_filt), results_transcripts)

head(results_transcripts)
```

```{r}
#sort results from the smallest p-value to the largest
results_transcripts = arrange(results_transcripts,pval)
head(results_transcripts)

results_genes = arrange(results_genes,pval)
head(results_genes)
```

```{r}
#output data to csv files
write.csv(results_transcripts, "chrX_transcript_results.csv",
row.names=FALSE)

write.csv(results_genes, "chrX_gene_results.csv",
row.names=FALSE)
```

```{r}
#identify transcripts and genes with a q-value less than 0.05
subset(results_transcripts,results_transcripts$qval<0.05)

subset(results_genes,results_genes$qval<0.05)
```

```{r}
#select colors for plotting
tropical= c('darkorange', 'dodgerblue',
'hotpink', 'limegreen', 'yellow')
palette(tropical)
```

```{r}
#access the FPKM data
fpkm = texpr(bg_chrX,meas="FPKM")

#log2 transform FPKM data, adding 1 to avoid "undefined" error 
fpkm = log2(fpkm+1)

#create boxplot of FPKM data
boxplot(fpkm,col=as.numeric(pheno_data$sex),las=2,ylab='log2(FPKM+1)')
```

```{r}
#create boxplot of log2 FPKM distrubitions of transcript NM_012227 for gene GTPBP6 in males and females

#display transcript name (version difference, GTPBP6 is transcript 11, not 12)
ballgown::transcriptNames(bg_chrX)[11]

#display gene name (version difference, GTPBP6 is transcript 11, not 12)
ballgown::geneNames(bg_chrX)[11]

#create boxplot
plot(fpkm[11,] ~ pheno_data$sex, border=c(1,2),
main=paste(ballgown::geneNames(bg_chrX)[11],' : ',
ballgown::transcriptNames(bg_chrX)[11]),pch=19, xlab="Sex",
ylab='log2(FPKM+1)')

#add points to plot
points(fpkm[11,] ~ jitter(as.numeric(pheno_data$sex)),
col=as.numeric(pheno_data$sex))



```

```{r}
#create plot of the structure and expression levels (FPKM) of XIST in sample ERR188234

#version difference, XIST is transcript 1659, not 1729
ballgown::geneNames(bg_chrX)[1659]

#create plot
plotTranscripts(ballgown::geneIDs(bg_chrX)[1659], bg_chrX, main=c('Gene XIST in
sample ERR188234'), sample=c('ERR188234'))
```

```{r}
#create plots of the structure and expression levels (FPKM) of MSTRG.56 in males and females
plotMeans('MSTRG.56', bg_chrX_filt, groupvar="sex", legend=FALSE)
```

